src/storage.zig

```diff
@@ -93,22 +94,21 @@ pub fn StorageType(comptime IO: type, comptime _Tracer: type) type {
 
         pub const NextTickSource = enum { lsm, vsr };
 
-        pub const Tracer = _Tracer;
-
         io: *IO,
         fd: IO.fd_t,
-        tracer: ?*Tracer = null,
+        tracer: *Tracer,
 
         next_tick_queue: QueueType(NextTick) = QueueType(NextTick).init(.{
             .name = "storage_next_tick",
         }),
         next_tick_completion_scheduled: bool = false,
         next_tick_completion: IO.Completion = undefined,
 
-        pub fn init(io: *IO, fd: IO.fd_t) !Storage {
+        pub fn init(io: *IO, fd: IO.fd_t, tracer: *Tracer) !Storage {
```

Comment:
```suggestion
        pub fn init(io: *IO, tracer: *Tracer, fd: IO.fd_t) !Storage {
```

`tracer` is a dependency, a context, like an `io` or an `allocator`. `fd` is a parameter that actually changes what the structure in question does. We generally put context first, then parameters, than callbacks last. 

Between `io` and `tracer`, `io` is more general, and should go first.